llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -14435,7 +14435,8 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
     // patterns on rv32..
     ConstantSDNode *Const = dyn_cast<ConstantSDNode>(Scalar);
     if (isOneConstant(VL) && EltWidth <= Subtarget.getXLen() &&
-        (!Const || Const->isZero() || !isInt<5>(Const->getSExtValue())))
+        (!Const || Const->isZero() ||
```

Comment:
Why is that SimplifyDemandedLowBitsHelper just above can't catch this case?  If only the low bits are demanded - which I think is what you're suggesting here - shouldn't that have folded the constant?

Reply:
I don't think the DemandedBits will turn a zero extended i64 constant into a sign extended i64 constant if the upper bits aren't used. Is that what you're suggesting?

Reply:
That was the question I'd asked, but to reframe it, why not go ahead and convert the constant to the sign extended form eagerly?  Even if we don't convert to the vmv_s_x, isn't that a reasonable canonicalization of the vmv_v_x?  

p.s. I'm completely fine with this landing as is, just suggesting a potentially cleaner way of doing it.